dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -566,21 +567,21 @@ public boolean checkAndSetDefaultVersion(String newDefaultVersion) {
     }
 
     /**
-     * Given a path (A/B/C/D), splits it into parts and returns it
+     * Given a path (A/B/C/D), splits it into parts and returns it.
+     * Need to indicate whether the path has an entry name in order to determine the repo name if there are more than 3 components in the path
      *
      * @param path
-     * @return An array of fields used to identify an entry
+     * @param hasEntryName boolean indicating whether the path has an entry name
+     * @return An array of fields used to identify components of an entry's path
      */
-    public static String[] splitPath(String path) {
-        // Used for accessing index of path
+    public static String[] splitPath(String path, boolean hasEntryName) {
```

Comment:
Unit tests for this method would be nice -- it's hard to tell at a glance if all paths/inputs work.